### PR TITLE
ui/get_path_length_idx:  line_x[i] shoud be less than or equal to path_height

### PR DESCRIPTION
**Pull request by deanlee**
the condition should be less than or equal to `path_height`,  otherwise the last valid point may be dropped incorrectly.

### CODE REVIEW

**Comment by deanlee** [comment]
the condition should be less than or equal to `path_height`,  otherwise the last valid point will be dropped incorrectly.

File: selfdrive/ui/ui.cc [hunk 1]
```diff
@@ -36,7 +36,7 @@ static bool calib_frame_to_full_frame(const UIState *s, float in_x, float in_y,
 static int get_path_length_idx(const cereal::ModelDataV2::XYZTData::Reader &line, const float path_height) {
   const auto line_x = line.getX();
   int max_idx = 0;
-  for (int i = 0; i < TRAJECTORY_SIZE && line_x[i] < path_height; ++i) {
+  for (int i = 1; i < TRAJECTORY_SIZE && std::islessequal(line_x[i], path_height); ++i) {
```
Comment (pd0wm) [inline selfdrive/ui/ui.cc:39]: Why `std::islessequal` instead of `<=`?

Comment (deanlee) [reply selfdrive/ui/ui.cc:39]: it's not very safe to do floating comparison with ==. the right way is `std::abs(a, b) < std::numeric_limits<float>::epsilon`. or use std::islessequal in this case.

Comment (deanlee) [reply selfdrive/ui/ui.cc:39]: I was worried that  the built-in operator<= may raise FE_INVALID if line_x[i] is NaN. but it seems unnecessary for now.

**Pull request merged**